### PR TITLE
Fix Cocoapods for Xcode 14.3.0

### PR DESCRIPTION
**Pull request by cipolleschi**
Summary:
On Thursday the 30th, Apple Released Xcode 14.3.0.
This Version of Xcode enforce some version checks for which React-Codegen, which supported iOS 11 as minimum supported version, could not be build anymore.

This change ensue that React-Codegen is always aligned to the min version supported by React Native.
Plus, it moves CircleCI's Xcode to 14.3.0, to keep this problem in Check.

While working on this, I figured that, with the monorepo, Ruby tests stopped working because they were in the wrong folder: I moved them in the right one.

## Changelog:
[iOS][Fixed] - Make React Native build with Xcode 14.3.0 and fix tests

Differential Revision: D44605617

### CODE REVIEW

File: packages/react-native/scripts/cocoapods/__tests__/codegen_utils-test.rb [hunk 1]
```diff
@@ -527,7 +535,7 @@ def get_podspec_no_fabric_no_script
           'source' => { :git => '' },
           'header_mappings_dir' => './',
           'platforms' => {
-            'ios' => '11.0',
+            'ios' => '12.4',
```
Comment (oferRounds) [inline packages/react-native/scripts/cocoapods/__tests__/codegen_utils-test.rb:538]: shouldn’t it be also `min_ios_version_supported`?

Comment (cipolleschi) [reply packages/react-native/scripts/cocoapods/__tests__/codegen_utils-test.rb:538]: it depends on the philosophy you adopt to write tests.
If you use `min_ios_version_supported` here, once we move the function to the proper file and we reuse the same function of production, this test would be tautological, as changing what the function in prod returns will also change what returned here.

Personally, I prefer to see my test failing if I bump a version in production code, so I can double check that the change was done on purpose and not by mistake. That's why I hardcoded the value.

Does it make sense?

Comment (oferRounds) [reply packages/react-native/scripts/cocoapods/__tests__/codegen_utils-test.rb:538]: interesting – makes sense!

**Comment by LukeIKG** [comment]
hey I'm new to developing. my react native project stopped working because of the xcode update. so my question is where do i look for updates? do i need to wait for an xcode update? Cocoapods update? I don't really understand what's happening.

Thanks for your help!

**Comment by Jonatthu** [comment]
@LukeIKG you need to wait until this gets merged or apply the changes manually with patch or downgrade your Xcode to 14.2

**Comment by LukeIKG** [comment]
> @LukeIKG you need to wait until this gets merged or apply the changes manually with patch or downgrade your Xcode to 14.2

Thanks for the quick response!
This means I'll wait for a new release and then update my package.json (at  "react-native": "0.71.4",) and run npm install -> pod install?

**Comment by cipolleschi** [comment]
So, the root cause is that one of our pods (React-Codegen) is supporting iOS 11, which does not build with Xcode 14.3.

This change bumps the above mentioned pod, fixing the problem.

When the patch is out, yes, you'll have to bump RN in the package.json, apply the changes suggested [here](https://react-native-community.github.io/upgrade-helper/) and then `yarn && cd ios && bundle exec pod install` - we suggest to use Ruby Bundler to install your dependencies as it makes sure to use the right version of Cocoapods. If you never used it before, the first time you need to run also `bundle install` inside the `iOS` folder. You need to run `bundle install` only once per project

Meanwhile, you can use the "patch" suggested [here](https://github.com/travis-mark/lrn/commit/015854716feadd61a904d5a603b027426472f863)

Or, I think, use Xcode 14.2 toolchain with Xcode 14.3, or downgrade Xcode to 14.2.

**Comment by LukeIKG** [comment]
> So, the root cause is that one of our pods (React-Codegen) is supporting iOS 11, which does not build with Xcode 14.3.
> 
> This change bumps the above mentioned pod, fixing the problem.
> 
> When the patch is out, yes, you'll have to bump RN in the package.json, apply the changes suggested [here](https://react-native-community.github.io/upgrade-helper/) and then `yarn && cd ios && bundle exec pod install` - we suggest to use Ruby Bundler to install your dependencies as it makes sure to use the right version of Cocoapods. If you never used it before, the first time you need to run also `bundle install` inside the `iOS` folder. You need to run `bundle install` only once per project
> 
> Meanwhile, you can use the "patch" suggested [here](https://github.com/travis-mark/lrn/commit/015854716feadd61a904d5a603b027426472f863)
> 
> Or, I think, use Xcode 14.2 toolchain with Xcode 14.3, or downgrade Xcode to 14.2.

Got it to work, yay!
Thanks a lot for clarifying the procedure - really helpful

**Comment by Jackyeoh** [comment]
> So, the root cause is that one of our pods (React-Codegen) is supporting iOS 11, which does not build with Xcode 14.3.
> 
> This change bumps the above mentioned pod, fixing the problem.
> 
> When the patch is out, yes, you'll have to bump RN in the package.json, apply the changes suggested [here](https://react-native-community.github.io/upgrade-helper/) and then `yarn && cd ios && bundle exec pod install` - we suggest to use Ruby Bundler to install your dependencies as it makes sure to use the right version of Cocoapods. If you never used it before, the first time you need to run also `bundle install` inside the `iOS` folder. You need to run `bundle install` only once per project
> 
> Meanwhile, you can use the "patch" suggested [here](https://github.com/travis-mark/lrn/commit/015854716feadd61a904d5a603b027426472f863)
> 
> Or, I think, use Xcode 14.2 toolchain with Xcode 14.3, or downgrade Xcode to 14.2.

For devs working with older versions of react-native (am working with 0.70.6) looking to apply the "patch", looks like we need to define min_ios_version_supported ourselves. The patch then looks like the following:

```
require_relative '../node_modules/react-native/scripts/react_native_pods'
require_relative '../node_modules/@react-native-community/cli-platform-ios/native_modules'

$min_ios_version_supported = 12.4

platform :ios, $min_ios_version_supported
install! 'cocoapods', :deterministic_uuids => false

# Force pods to match minimum iOS version for React Native
# Fixes build issue on Xcode Cloud where some pods
# Use iOS 12 calls despite being set as iOS 11
def __apply_Xcode_14_3_RC_post_install_workaround(installer)
  installer.pods_project.targets.each do |target|
    target.build_configurations.each do |config|
      current_target = config.build_settings['IPHONEOS_DEPLOYMENT_TARGET']
      minimum_target = $min_ios_version_supported
      if current_target.to_f < minimum_target.to_f
        config.build_settings['IPHONEOS_DEPLOYMENT_TARGET'] = minimum_target
      end
    end
  end
end
```

**Comment by Shakahs** [comment]
So can we infer that Meta and the React Native project do not test XCode betas to prevent this kind of toolchain breakage?

**Comment by guilepaul** [comment]
Just change min_ios_version_supported in Xcode for 12.4
Captura de Tela 2023-04-03 às 15 24 23

**Comment by flochtililoch** [comment]
I'm a little confused with this statement:

> the root cause is that one of our pods (React-Codegen) is supporting iOS 11, which does not build with Xcode 14.3

Xcode 14.3 can build for iOS 11, so I assume that there's some code in `React-Codegen` that does not play nice with Xcode 14.3 and that it somewhat relates to iOS 11 support. 

@cipolleschi could you provide more clarity on what is causing the build issues?

I'm currently running an older RN version (0.67.4), which does have a minimum iOS target version of 11. I'd like to know if bumping it to 12.4 is the only viable solution right now to be able to build with Xcode 14.3

**Pull request closed without merging**